client/jetpack-connect/signup.js

```diff
@@ -242,12 +394,20 @@ export class JetpackSignup extends Component {
 	}
 }
 
-const connectComponent = connect( null, {
-	createAccount: createAccountAction,
-	createSocialAccount: createSocialAccountAction,
-	errorNotice: errorNoticeAction,
-	recordTracksEvent: recordTracksEventAction,
-	warningNotice: warningNoticeAction,
-} );
+const connectComponent = connect(
+	( state ) => ( {
+		requestError: getRequestError( state ),
+		usernameOrEmail: getLastCheckedUsernameOrEmail( state ),
+		isFullLoginFormVisible: !! getAuthAccountType( state ),
```

Comment:
What's the significance of the auth account type to whether or not the full login form is visible?

Reply:
See my reply [here](https://github.com/Automattic/wp-calypso/pull/41798#discussion_r432938870). Basically, the initial state is:
- Account type is not set (the user hasn't submitted a username/email to check its status)
- The only field visible is the Username/email field (as shown in the designs).

Then, when the user inputs an email and clicks the submit button, a request is made to the server. If there's an account with that email, the state will be:
- Account type will be set, it will be either `regular` or `passwordless` (not sure if these are the actual values, but that's not important right now).
- As such, at this point the user has the username/email field prefilled, and the rest of the login form (the password field) must appear.